src/app/core/services/asset-property-mapper.ts

```diff
@@ -27,10 +27,11 @@ export class AssetPropertyMapper {
     private activeFeatureSet: ActiveFeatureSet,
   ) {}
 
-  buildAssetFromProperties(
-    props: Record<string, string | null>,
-    opts?: {connectorEndpoint?: string},
-  ): Asset {
+  buildAsset(opts: {
+    connectorEndpoint: string;
+    properties: Record<string, string | null>;
+  }): Asset {
+    const props = opts.properties;
     const language = props[AssetProperties.language]
       ? this.languageSelectItemService.findById(
           props[AssetProperties.language]!,
```

Comment:
The same logic is being used for 4 properties, make sense to use a helper function?